storage/ksql/src/main/java/io/apicurio/registry/storage/impl/ksql/serde/KafkaSqlValueSerializer.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.storage.impl.ksql.serde;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.ByteBuffer;
+
+import org.apache.commons.io.output.UnsynchronizedByteArrayOutputStream;
+import org.apache.kafka.common.serialization.Serializer;
+
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.apicurio.registry.storage.impl.ksql.keys.MessageType;
+import io.apicurio.registry.storage.impl.ksql.values.ContentValue;
+import io.apicurio.registry.storage.impl.ksql.values.MessageValue;
+
+/**
+ * Responsible for serializing the message key to bytes.
+ * @author eric.wittmann@gmail.com
+ */
+public class KafkaSqlValueSerializer implements Serializer<MessageValue> {
+    
+    private static final ObjectMapper mapper = new ObjectMapper();
+    static {
+        mapper.setSerializationInclusion(Include.NON_NULL);
+    }
+
+    /**
+     * @see org.apache.kafka.common.serialization.Serializer#serialize(java.lang.String, java.lang.Object)
+     */
+    @Override
+    public byte[] serialize(String topic, MessageValue messageValue) {
+        if (messageValue.getType() == MessageType.Content) {
+            return this.serializeContent(topic, (ContentValue) messageValue);
+        }
+        try (UnsynchronizedByteArrayOutputStream out = new UnsynchronizedByteArrayOutputStream()) {
+            out.write(ByteBuffer.allocate(1).put((byte) messageValue.getType().ordinal()).array());
+            mapper.writeValue(out, messageValue);
```

Comment:
this is not protobuf nor avro :) but it's ok , we could even leave it like this but the message keys... that is another story

Reply:
Yeah, my intention was to solicit a volunteer to re-implement the Serdes (both key and value) to use something like protobuf.  I felt like that could easily be done in a followup PR.

Reply:
I will create a GH issue for this: https://github.com/Apicurio/apicurio-registry/issues/1059